source/gui/settingsDialogs.py

```diff
@@ -1160,24 +1160,30 @@ def makeSettings(self, settingsSizer):
 		except:
 			log.debugWarning("Could not set Keyboard layout list to current layout",exc_info=True) 
 
-		# Translators: This is the label for a checkbox in the
-		# keyboard settings panel.
-		capsAsNVDAText = _("Use CapsLock as an NVDA modifier key")
-		self.capsAsNVDAModifierCheckBox=sHelper.addItem(wx.CheckBox(self,label=capsAsNVDAText))
-		self.capsAsNVDAModifierCheckBox.SetValue(config.conf["keyboard"]["useCapsLockAsNVDAModifierKey"])
-
-		# Translators: This is the label for a checkbox in the
-		# keyboard settings panel.
-		numpadInsertAsModText = _("Use numpad Insert as an NVDA modifier key")
-		self.numpadInsertAsNVDAModifierCheckBox=sHelper.addItem(wx.CheckBox(self,label=numpadInsertAsModText))
-		self.numpadInsertAsNVDAModifierCheckBox.SetValue(config.conf["keyboard"]["useNumpadInsertAsNVDAModifierKey"])
-
-		# Translators: This is the label for a checkbox in the
-		# keyboard settings panel.
-		extendedInsertAsModText = _("Use extended Insert as an NVDA modifier key")
-		self.extendedInsertAsNVDAModifierCheckBox=sHelper.addItem(wx.CheckBox(self,label=extendedInsertAsModText))
-		self.extendedInsertAsNVDAModifierCheckBox.SetValue(config.conf["keyboard"]["useExtendedInsertAsNVDAModifierKey"])
-
+		#Translators: This is the label for a list of checkboxes
+		# controlling which keys are NVDA modifier keys.
+		modifierBoxLabel = _("&Select NVDA Modifier Keys")
+		self.modifierChoices = [

```

Comment:
Personally, I"d do this similar to how focusContextPresentation is implemented in the gui.

1. Store the possible modifiers in a list on keyboardHandler, just the key names
2. Rather than making the names translatable, use keyLabels.localizedKeyLabels in the gui

Of course you could also store the possible modifiers around this line. Then, when saving, use modifierslist.index("insert") in self.kbdList.Checked.

You could also consider changing the config spec into a string list. Though it would require a config spec upgrade, it would create a solid reference from which people could start implementing another checkable list.